diff --git a/client_v2/src/common/ui/Menu/Menu.tsx b/client_v2/src/common/ui/Menu/Menu.tsx
index ae74aa1..bbf8487 100644
--- a/client_v2/src/common/ui/Menu/Menu.tsx
+++ b/client_v2/src/common/ui/Menu/Menu.tsx
@@ -127,17 +127,6 @@ export const Menu = (props: Props) => {
                         <span class={theme.common.textOverflow}>{intl.getMessage('logs')}</span>
                     </Link>
                 </div>
-                <div class={cn(s.menuLinkWrapper)}>
-                    <Link
-                        class={cn(s.menuLink, { [s.activeLink]: isActive(Paths.Guide) })}
-                        to={RoutePath.Guide}
-                    >
-                        <Icon class={s.linkIcon} icon="faq" />
-                        <span class={theme.common.textOverflow}>
-                            {intl.getMessage('setup_guide')}
-                        </span>
-                    </Link>
-                </div>
             </nav>
             <div class={s.referenceWrapper}>
                 <div class={cn(s.menuLinkWrapper)}>
